xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -577,6 +596,12 @@ where
         envelope: &GroupMessageV1,
         allow_cursor_increment: bool,
     ) -> Result<(), GroupMessageProcessingError> {
+        #[cfg(any(test, feature = "test-utils"))]
```

Comment:
can probably reduce this to a nice two-liner:
```suggestion
        #[cfg(any(test, feature = "test-utils"))]
        crate::utils::maybe_mock_wrong_epoch_for_tests()?;
```